### PR TITLE
Update to current deal.II version

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: common/multigrid_solver.h [hunk 1]
```diff
@@ -672,7 +672,11 @@ namespace multigrid
           v_cycle(level-1, 1);
 
           time.restart();
+#if false
           transfer.prolongate_add(level, solution_update[level], solution_update[level-1]);
+#else
+          transfer.prolongate(level, solution_update[level], solution_update[level-1]);
+#endif
```
Comment (peterrum) [inline common/multigrid_solver.h]: TODO

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+ADD_SUBDIRECTORY(cmake)
+
+ADD_SUBDIRECTORY(matvec_dg)
+ADD_SUBDIRECTORY(matvec_dg_cheby)
+#ADD_SUBDIRECTORY(minimal_surface)
+ADD_SUBDIRECTORY(poisson_cube)
+ADD_SUBDIRECTORY(poisson_dg)
+ADD_SUBDIRECTORY(poisson_dg_plain)
+#ADD_SUBDIRECTORY(poisson_l)
```
Comment (peterrum) [inline CMakeLists.txt:59]: I was not able to get these two programs running.

**Review by peterrum — COMMENTED** [review]
@kronbichler This should be all. It's amazing how much has changed in deal.II in the meantime...

**Comment by peterrum** [comment]
I am having problems running `poisson_dg_plain`:
```
An error occurred in line <579> of file </home/munch/sw-index/multigrid/poisson_dg_plain/../common/laplace_operator_dg.h> in function
    void multigrid::LaplaceOperatorCompactCombine<dim, fe_degree, Number, type>::reinit(std::shared_ptr<const dealii::MatrixFree<dim, number> >, unsigned int, const multigrid::LaplaceOperator<dim, fe_degree, Number>*) [with int dim = 3; int fe_degree = 5; Number = float; int type = 0]
The violated condition was: 
    matrixfree->get_mapping_info().cell_data[0].jacobians[0].size()<=1
Additional information: 
    You are trying to use functionality in deal.II that is currently not
    implemented. In many cases, this indicates that there simply didn't
    appear much of a need for it, or that the author of the original code
    did not have the time to implement a particular case. If you hit this
    exception, it is therefore worth the time to look into the code to
    find out whether you may be able to implement the missing
    functionality. If you do, please consider providing a patch to the
    deal.II development sources (see the deal.II website on how to
    contribute).
```
I would interpret the error that not an affine mesh is given!?

**Comment by kronbichler** [comment]
I think you need to change the check to `<= 2` because we nowadays also store the Jacobian and not just the inverse in this array. This is needed for the way we compute the quadrature points.

**Comment by kronbichler** [comment]
> It's amazing how much has changed in deal.II in the meantime...

This is why I am always try to avoid too many incompatible changes or deprecation of too much functionality - it is a huge pile of work in every downstream project.

File: CMakeLists.txt (hunk 2 above)
Comment (kronbichler) [reply CMakeLists.txt:59]: They should be running - I will need to have a look.

Comment (kronbichler) [reply CMakeLists.txt:59]: I need to postpone it to #3.

**Comment by peterrum** [comment]
> I think you need to change the check to <= 2 because we nowadays also store the Jacobian and not just the inverse in this array. This is needed for the way we compute the quadrature points.

Thanks! That was it!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**